client/chain.go

```diff
@@ -117,349 +164,161 @@ func (c *chainClient) syncNonce() {
+	return c.conn
+}
 
-		if initSeq == 0 {
-			txf = txf.WithSequence(seq)
-		}
-	}
+func (c *chainClient) ClientContext() client.Context {
+	return c.ctx
+}
 
-	return txf, nil
+func (c *chainClient) CanSignTransactions() bool {
+	return c.canSign
 }
 
-// QueueBroadcastMsg enqueues a list of messages. Messages will added to the queue
-// and grouped into Txns in chunks. Use this method to mass broadcast Txns with efficiency.
-func (c *chainClient) QueueBroadcastMsg(msgs ...sdk.Msg) error {
+func (c *chainClient) FromAddress() sdk.AccAddress {
 	if !c.canSign {
-		return ErrReadOnly
-	} else if atomic.LoadInt64(&c.closed) == 1 {
-		return ErrQueueClosed
-	}
-
-	t := time.NewTimer(10 * time.Second)
-	for _, msg := range msgs {
-		select {
-		case <-t.C:
-			return ErrEnqueueTimeout
-		case c.msgC <- msg:
-		}
+		return sdk.AccAddress{}
 	}
-	t.Stop()
 
-	return nil
+	return c.ctx.FromAddress
 }
 
 func (c *chainClient) Close() {
 	if !c.canSign {
 		return
 	}
-
 	if atomic.CompareAndSwapInt64(&c.closed, 0, 1) {
 		close(c.msgC)
 	}
-
 	<-c.doneC
-
 	if c.conn != nil {
 		c.conn.Close()
 	}
 }
 
-const (
-	msgCommitBatchSizeLimit = 1024
-	msgCommitBatchTimeLimit = 500 * time.Millisecond
-)
-
-func (c *chainClient) runBatchBroadcast() {
-	expirationTimer := time.NewTimer(msgCommitBatchTimeLimit)
-	msgBatch := make([]sdk.Msg, 0, msgCommitBatchSizeLimit)
-
-	submitBatch := func(toSubmit []sdk.Msg) {
-		c.syncMux.Lock()
-		defer c.syncMux.Unlock()
-
-		c.txFactory = c.txFactory.WithSequence(c.accSeq)
-		c.txFactory = c.txFactory.WithAccountNumber(c.accNum)
-		log.Debugln("broadcastTx with nonce", c.accSeq)
-		res, err := c.broadcastTx(c.ctx, c.txFactory, true, toSubmit...)
-		if err != nil {
-			if strings.Contains(err.Error(), "account sequence mismatch") {
-				c.syncNonce()
-				c.txFactory = c.txFactory.WithSequence(c.accSeq)
-				c.txFactory = c.txFactory.WithAccountNumber(c.accNum)
-				log.Debugln("retrying broadcastTx with nonce", c.accSeq)
-				res, err = c.broadcastTx(c.ctx, c.txFactory, true, toSubmit...)
-			}
-			if err != nil {
-				resJSON, _ := json.MarshalIndent(res, "", "\t")
-				c.logger.WithField("size", len(toSubmit)).WithError(err).Errorln("failed to commit msg batch:", string(resJSON))
-				return
-			}
-		}
-
-		if res.Code != 0 {
-			err = errors.Errorf("error %d (%s): %s", res.Code, res.Codespace, res.RawLog)
-			log.WithField("txHash", res.TxHash).WithError(err).Errorln("failed to commit msg batch")
-		} else {
-			log.WithField("txHash", res.TxHash).Debugln("msg batch committed successfully")
-		}
+func (c *chainClient) BlockBroadcastTx(msgs ...sdk.Msg) (*txtypes.BroadcastTxResponse, error) {
+	return c.broadcastTx(txtypes.BroadcastMode_BROADCAST_MODE_BLOCK, msgs...)
+}
 
-		c.accSeq++
-		log.Debugln("nonce incremented to", c.accSeq)
-	}
+func (c *chainClient) SyncBroadcastTx(msgs ...sdk.Msg) (*txtypes.BroadcastTxResponse, error) {
+	return c.broadcastTx(txtypes.BroadcastMode_BROADCAST_MODE_SYNC, msgs...)
+}
 
-	for {
```

Comment:
I don't like that the batching code is being scrapped completely. It was added on purpose, resolving the issues in the default mechanism provided by Cosmos SDK, which doesn't support more throughput than 1 tx per block. Can you please document the rationale that is going behind these changes? Including all the drawbacks that this may impose.

Also, what was the original issue?

Reply:
The sdk-go client has been built to be bullet-proof and was thoroughly used for many months now, what's the particular reason to change it?